### PR TITLE
Adapt resources list to project storage provider

### PR DESCRIPTION
**Pull request by AlexandreSi**
Cloud project on desktop app
image

Local project on desktop app
image

Cloud project on web app
image

### CODE REVIEW

File: newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectResourcesHandler.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+}: {
+  authenticatedUser: AuthenticatedUser,
+}) => ({
+  project,
+  resource,
+  i18n,
+  fileMetadata,
+  informUser,
+}: ResourcesActionsProps) => {
+  const openLabel = app && path ? t`Download` : t`Open resource in navigator`;
```
Comment (ClementPasteau) [inline newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectResourcesHandler.js:30]: open resource in browser instead?

File: newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectResourcesHandler.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        );
+        if (checkIfIsGDevelopCloudBucketUrl(resourceUrl)) {
+          await getCredentialsForCloudProject(
+            authenticatedUser,
+            fileMetadata.fileIdentifier
+          );
+        }
+        if (app && path && electron) {
+          const targetPath = path.join(
+            app.getPath('downloads'),
```
Comment (ClementPasteau) [inline newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectResourcesHandler.js:48]: is there a chance the user has no write access on this folder?
If so, should we try catch to show an error? Or have a fallback?

**Review by ClementPasteau — APPROVED** [review]
nice

File: newIDE/app/src/ResourcesList/index.js [hunk 3]
```diff
@@ -405,6 +318,14 @@ export default class ResourcesList extends React.Component<Props, State> {
             )}
           </AutoSizer>
         </div>
+        {!!infoBarContent && (
+          <InfoBar
+            duration={7000}
```
[the remark this answers is not in the document]
Comment (AlexandreSi) [reply newIDE/app/src/ResourcesList/index.js:323]: Yes

File: newIDE/app/src/ResourcesList/index.js [hunk 4]
```diff
@@ -405,6 +318,14 @@ export default class ResourcesList extends React.Component<Props, State> {
...
+            visible={!!infoBarContent}
```
[the remark this answers is not in the document]
Comment (AlexandreSi) [reply newIDE/app/src/ResourcesList/index.js:324]: I need the condition above so I'll set it to true, thanks

File: newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectResourcesHandler.js [hunk 5]
```diff
@@ -27,47 +27,77 @@ export const generateGetResourceActions = ({
   fileMetadata,
   informUser,
 }: ResourcesActionsProps) => {
-  const openLabel = app && path ? t`Download` : t`Open resource in browser`;
-  return [
+  const openLabel =
+    app && path
+      ? t`Download under "Downloads" folder`
```
Comment (ClementPasteau) [inline newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectResourcesHandler.js:32]: I'd go for `Download into "Downloads" folder`

Comment (ClementPasteau) [reply newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectResourcesHandler.js:32]: Or maybe `Save into "Downloads" folder`

**Pull request merged**